src/types.jl

```diff
@@ -7,14 +7,9 @@ struct Arf <: Real
         return res
     end
 
-    function Arf(x::arf_struct; prec::Integer = DEFAULT_PRECISION[], shallow::Bool = false)
-        if shallow
-            res = new(x, prec)
-        else
-            res = Arf(prec = prec)
-            set!(res, x)
-        end
-
+    function Arf(x::arf_struct; prec::Integer = DEFAULT_PRECISION[])
+        res = Arf(prec = prec)
+        set!(res, x)
```

Comment:
I see you have removed the `shallow` parameter here and in other places. I guess this is because we are now using `ArbRef` and `AcbRef`? Doesn't the `shallow` parameter still make sense if you want to copy, for example, an `Arb` only increasing the precision of it? I guess this might not be something you do in a tight loop in the same way as iterating through an `ArbVec` or `ArbMatrix`, so it might not be as important to optimise for though.

Reply:
Yes, I removed this since this primarily used for the vector / matrix indexing. I have some thoughts about the semantics of changing the precision, but I didn't want to drag this in here.